sdk/program/src/vote/state/mod.rs

```diff
@@ -333,8 +337,18 @@ impl VoteState {
                 // This is also to cancel the rewarding if either of the parties
                 // should receive only fractional lamports, resulting in not being rewarded at all.
                 // Thus, note that we intentionally discard any residual fractional lamports.
-                let mine = on * u128::from(split) / 100u128;
-                let theirs = on * u128::from(100 - split) / 100u128;
+                let mine = on
+                    .checked_mul(u128::from(split))
+                    .expect("multiplication of a u64 and u8 should not overflow")
+                    / 100u128;
+                let theirs = on
```

Comment:
aside: never understood why we don't just subtract here :upside_down_face:

Reply:
yeah not sold on the explanation, we're leaving fractional lamports behind 💸